File: order-service/src/main/java/com/myKart/order/config/OpenApiConfig.java
package com.mykart.order.config;

import io.swagger.v3.oas.annotations.OpenAPIDefinition;
import io.swagger.v3.oas.annotations.enums.SecuritySchemeType;
import io.swagger.v3.oas.annotations.info.Info;
import io.swagger.v3.oas.annotations.security.SecurityScheme;
import org.springframework.context.annotation.Configuration;

@Configuration
@OpenAPIDefinition(info = @Info(title = "Order Service API", version = "v2",
        description = "Order placement with Saga orchestration and Outbox pattern"))
@SecurityScheme(name = "bearerAuth", type = SecuritySchemeType.HTTP, scheme = "bearer", bearerFormat = "JWT")
public class OpenApiConfig {}

File: payment-service/src/main/kotlin/com/mykart/payment/config/OpenApiConfig.kt
package com.mykart.payment.config

import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class OpenApiConfig {

    @Bean
    fun openApi(): OpenAPI = OpenAPI().info(
        Info()
            .title("Payment Service API")
            .description("Seller payout disbursement — UPI / IMPS / NEFT channels")
            .version("1.0.0")
    )
}
